modules/offers/tests/server/offer-search.server.routes.tests.js

```diff
@@ -1147,6 +1147,135 @@ describe('Offer search tests', function () {
     });
   });
 
+  it('should be able to get offers from users with circles in common and have "showOnlyMyCircles" set', function (done) {
+    // Verify that offers where showOnlyInMyCircles is true are only appearing
+    // in searches where the authenticated user (user1) has at least one circle
+    // in common with the user that owns the offer.
+    //
+    // Makes the users members of the following tribes:
+    //   - user1: tribe1, tribe2  (the authenticated user)
+    //   - user2: tribe2, tribe3  (overlaps with user1)
+    //   - user3: tribe3          (does not overlap with user1)
+    // The following hosting offers are available:
+    //   - offer1 is owned by user2 and should appear as user1's circles overlap
+    //     with the ones of user2 while showOnlyInMyCircles is true.
+    //   - offer2 is owned by user3 and should *not* appear as user1's circles
+    //     do not overlap with the ones of user3 while showOnlyInMyCircles is true.
+    //   - offer3 is owned by user3 and should appear despite user1's circles not
+    //     overlapping with the ones of user3 since showOnlyInMyCircles is false.
+    let tribe3Id;
+    let offer1Id;
+    async.waterfall(
+      [
+        // Save tribe 3.
+        function (done) {
+          const tribe3 = new Tribe({
+            slug: 'tribe3',
+            label: 'tribe3',
+            color: '333333',
+            count: 1,
+            public: true,
+          });
+          tribe3.save(function (err, tribe3) {
+            tribe3Id = tribe3._id;
+            done(err);
+          });
+        },
+
+        // Set the users' memberships.
+        function (done) {
+          user1.member = [
+            { tribe: tribe1Id, since: new Date() },
+            { tribe: tribe2Id, since: new Date() },
+          ];
+          user1.save(done);
+        },
+        function (user1, done) {
+          user2.member = [
+            { tribe: tribe2Id, since: new Date() },
+            { tribe: tribe3Id, since: new Date() },
+          ];
+          user2.save(done);
+        },
+        function (user2, done) {
+          user3.member = { tribe: tribe3Id, since: new Date() };
+          user3.save(done);
+        },
+
+        // Update the hosting offers.
+        function (user3, done) {
+          // Save hosting offer 1 (user2, showOnlyInMyCircles=true).
+          const o1 = new Offer(offer1);
+          o1.user = user2Id;
+          o1.location = testLocations.Europe.location;
+          o1.showOnlyInMyCircles = true;
```

Comment:
An alternative way to write this would be:

```js
const o1 = new Offer({
  ...offer1,
  user: user2Id,
  location: testLocations.Europe.location,
  showOnlyInMyCircles: true,
});
```

Up to you!